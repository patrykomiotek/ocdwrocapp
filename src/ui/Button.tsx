// const buttonStyle: React.CSSProperties = {
//       backgroundColor: palette[bgColor],
//       color: palette[color],
//       padding: '10px 20px',
//       border: 'none',
//       borderRadius: '5px',
//       cursor: 'pointer',
//       fontSize: '16px',
//       fontWeight: 'bold',
//       transition: 'all 0.3s ease',
//     };

const palette = {
  turquoise: "#1abc9c",
  emerald: "#2ecc71",
  peterRiver: "#3498db",
  amethyst: "#9b59b6",
  wetAsphalt: "#34495e",
  greenSea: "#16a085",
  nephritis: "#27ae60",
  belizeHole: "#2980b9",
  wisteria: "#8e44ad",
  midnightBlue: "#2c3e50",
  sunFlower: "#f1c40f",
  carrot: "#e67e22",
  alizarin: "#e74c3c",
  clouds: "#ecf0f1",
  concrete: "#95a5a6",
  orange: "#f39c12",
  pumpkin: "#d35400",
  pomegranate: "#c0392b",
  silver: "#bdc3c7",
  asbestos: "#7f8c8d",
};

// enum Color {
//   alizarin = "#e74c3c",
//   clouds = "#ecf0f1",
// }

type Color = keyof typeof palette;

interface Props {
  children: string;
  color?: Color;
  bgColor?: Color;
}

export function Button({
  children,
  color = "clouds", // Color.alizarin
  bgColor = "carrot", // Color.clouds
}: Props) {
  const styles = { color: palette[color], backgroundColor: palette[bgColor] };

  return <button style={styles}>{children}</button>;
}

{
  /* <Button bgColor="clouds" color="carrot">Click</Button> */
  // <Button color="clouds">Click me</Button>
  // <Button>Click me</Button>
}
