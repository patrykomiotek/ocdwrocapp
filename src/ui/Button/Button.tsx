import { memo, type ComponentProps } from 'react';

const palette = {
  turquoise: '#1abc9c',
  emerald: '#2ecc71',
  peterRiver: '#3498db',
  amethyst: '#9b59b6',
  wetAsphalt: '#34495e',
  greenSea: '#16a085',
  nephritis: '#27ae60',
  belizeHole: '#2980b9',
  wisteria: '#8e44ad',
  midnightBlue: '#2c3e50',
  sunFlower: '#f1c40f',
  carrot: '#e67e22',
  alizarin: '#e74c3c',
  clouds: '#ecf0f1',
  concrete: '#95a5a6',
  orange: '#f39c12',
  pumpkin: '#d35400',
  pomegranate: '#c0392b',
  silver: '#bdc3c7',
  asbestos: '#7f8c8d',
};

// enum Color {
//   alizarin = "#e74c3c",
//   clouds = "#ecf0f1",
// }

type Color = keyof typeof palette;

interface Props extends ComponentProps<'button'> {
  // children: string;
  children: React.ReactNode;
  color?: Color;
  bgColor?: Color;
}

// type ButtonProps = Props & Required<Pick<Props, "onClick">>;
// type ButtonProps = Required<Props>; // all are required

export const Button = memo(
  ({
    children,
    color = 'clouds', // Color.alizarin
    bgColor = 'carrot', // Color.clouds
    ...rest
  }: Props) => {
    const buttonStyle: React.CSSProperties = {
      backgroundColor: palette[bgColor],
      color: palette[color],
      padding: '10px 20px',
      border: 'none',
      borderRadius: '5px',
      cursor: 'pointer',
      fontSize: '16px',
      fontWeight: 'bold',
      transition: 'all 0.3s ease',
    };

    return (
      <button {...rest} style={buttonStyle}>
        {children}
      </button>
    );
  },
);

Button.displayName = 'memo(Button)';

{
  /* <Button bgColor="clouds" color="carrot">Click</Button> */
  // <Button color="clouds">Click me</Button>
  // <Button>Click me</Button>
}
