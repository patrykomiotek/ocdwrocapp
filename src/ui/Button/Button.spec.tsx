import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Button } from './Button';

describe('Button component', () => {
  it('worst test ever', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Button>Click me</Button>);

    expect(container).toMatchSnapshot();
  });

  it('should pass accessibility check', async () => {
    const { container } = render(<Button>Click me</Button>);

    const result = await axe(container);

    expect(result).toHaveNoViolations();
  });
});
