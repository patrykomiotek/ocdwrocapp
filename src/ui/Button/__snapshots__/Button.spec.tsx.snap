// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should match snapshot 1`] = `
<div>
  <button
    style="background-color: rgb(230, 126, 34); color: rgb(236, 240, 241); padding: 10px 20px; border-radius: 5px; cursor: pointer; font-size: 16px; font-weight: bold; transition: all 0.3s ease;"
  >
    Click me
  </button>
</div>
`;
